tests/test_package.py

```diff
@@ -381,76 +387,64 @@ def test_fips_metadata_excludes_md5_and_blake2(monkeypatch):
+    "read_data, exception_message",
     [
         pytest.param(
-            b"Metadata-Version: 2.3\nName: UNKNOWN\nVersion: UNKNOWN\n",
-            "Name, Version",
+            b"Metadata-Version: 102.3\nName: test-package\nVersion: 1.0.0\n",
+            "'102.3' is not a valid metadata version",
+            id="unsupported Metadata-Version",
+        ),
+        pytest.param(
+            b"Metadata-Version: 2.3\nName: test-package\nVersion: UNKNOWN\n",
+            "'UNKNOWN' is invalid for 'version'",
+            id="invalid Version",
+        ),
+        pytest.param(
+            b"Metadata-Version: 2.2\nName: test-package\nVersion: UNKNOWN\n",
+            "'UNKNOWN' is invalid for 'version'",
+            id="invalid Version",
+        ),
+        pytest.param(
+            b"Metadata-Version: 2.3\n",
+            "'name' is a required field, 'version' is a required field",
             id="missing Name and Version",
         ),
         pytest.param(
-            b"Metadata-Version: 2.2\nName: UNKNOWN\nVersion: UNKNOWN\n",
-            "Name, Version",
+            b"Metadata-Version: 2.2\n",
+            "'name' is a required field, 'version' is a required field",
             id="missing Name and Version",
         ),
         pytest.param(
-            b"Metadata-Version: 2.3\nName: UNKNOWN\nVersion: 1.0.0\n",
-            "Name",
+            b"Metadata-Version: 2.3\nVersion: 1.0.0\n",
+            "'name' is a required field",
             id="missing Name",
         ),
         pytest.param(
-            b"Metadata-Version: 2.2\nName: UNKNOWN\nVersion: 1.0.0\n",
-            "Name",
+            b"Metadata-Version: 2.2\nVersion: 1.0.0\n",
+            "'name' is a required field",
             id="missing Name",
         ),
         pytest.param(
-            b"Metadata-Version: 2.3\nName: test-package\nVersion: UNKNOWN\n",
-            "Version",
+            b"Metadata-Version: 2.3\nName: test-package\n",
+            "'version' is a required field",
             id="missing Version",
         ),
         pytest.param(
-            b"Metadata-Version: 2.2\nName: test-package\nVersion: UNKNOWN\n",
-            "Version",
+            b"Metadata-Version: 2.2\nName: test-package\n",
+            "'version' is a required field",
             id="missing Version",
         ),
     ],
 )
-def test_pkginfo_returns_no_metadata(read_data, missing_fields, monkeypatch):
+def test_pkginfo_returns_no_metadata(read_data, exception_message, monkeypatch):
     """Raise an exception when pkginfo can't interpret the metadata."""
     monkeypatch.setattr(package_file.wheel.Wheel, "read", lambda _: read_data)
     filename = "tests/fixtures/twine-1.5.0-py2.py3-none-any.whl"
 
     with pytest.raises(exceptions.InvalidDistribution) as err:
         package_file.PackageFile.from_filename(filename, comment=None)
 
-    assert (
-        f"Metadata is missing required fields: {missing_fields}." in err.value.args[0]
-    )
```

Comment:
The exception message is simply a concatenation of the validation errors returned by `packaging.metadata.Metadata.from_raw()`. I can change how they are concatenated, but it is unrealistic to change them otherwise.